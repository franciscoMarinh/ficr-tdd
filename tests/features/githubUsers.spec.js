const { get } = require('../e2e/utils/api');
const nock = require('nock');
const githubSuccessResponse = require('../fixtures/github-users-response.json');
const githubNotFoundResponse = require('../fixtures/github-users-notfound.json');

describe('Github fetch Users', () => {
	test('deverÃ¡ retornar o usuario que exista', async () => {
		const username = 'franciscoMarinh';

		nock('https://api.github.com')
			.get(`/users/${username}`)
			.reply(200, githubSuccessResponse);

		const res = await get(`/api/github/${username}`);

		expect(res.status).toEqual(200);

		expect(res.body).toMatchSnapshot({
			nome: githubSuccessResponse.name,
			bio: githubSuccessResponse.bio,
			githubPage: githubSuccessResponse.html_url,
		});
	});

	test('Deve retornar mensagem de que o usuario nao existe', async () => {
		const username = 'userNotFound';

		nock('https://api.github.com')
			.get(`/users/${username}`)
			.reply(404, githubNotFoundResponse);

		const res = await get(`/api/github/${username}`);

		expect(res.status).toEqual(404);
	});
});
