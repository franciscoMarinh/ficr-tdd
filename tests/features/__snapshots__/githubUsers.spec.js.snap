// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Github fetch Users dever√° retornar o usuario que exista 1`] = `
Object {
  "bio": "Graduating in systems analysis and development",
  "githubPage": "https://github.com/franciscoMarinh",
  "nome": "Francisco marinho",
}
`;
